src/vision/status_quo/barbara_wants_async_insights.md

```diff
@@ -0,0 +1,70 @@
+
+Once she has the program compiling, she thinks "oh that was easy". she runs it for the first time and unfortunately when hitting an endpoint, nothing happens.
+
+Barbara, always prepared, has already added logging to her service and she checks the logs. As she expected, she sees here that the endpoint handler has been invoked but then... nothing. Barbara exclaims, "Oh no! This was not what I was expecting, but let's dig deeper." 
+
+She checks the code and sees that the endpoint spawns several tasks, but unfortunately those tasks don't have much logging in them.
+
+Barbara knows that debugging with a traditional debugger is not very fruitful in async Rust. She does a deep dive into the source code and doesn't find anything. Then she adds much more logging, but to her dismay she finds that a particular task seems stuck, but she has no idea why.
+
+She really wishes that there was a way to get more insight into why the task is stuck. These were the thoughts inside her head at that moment:
+* Is it waiting on I/O?
+* Is there a deadlock?
+* Did she miss some sync code that might still be there and messing with the executor?
+
+For the I/O question she knows to use some tools on her operating system (lsof). This reveals some open sockets but she's not sure how to act on this.
+
+She scans the code for any std lib imports that might be blocking, but doesn't find anything.
+
+After hours of crawling through the code, she notices that her task is receiving a message from a bounded async channel. She changes this to be an unbounded channel and then things start working.
+
+She wants to know why the code was not working, but unfortunately she has no way to gain insight into this issue. She fears that her task might use too much memory knowing that the channel is unbounded, but she can't really tell.
+
+She thinks, "Anyhow it is working now, let's see if we got some performance gains." After thorough benchmarking she finds out that she didn't quite get the performance gain she was expecting. "Something is not working, as intended", she thinks.
+
+## 🤔 Frequently Asked Questions
+
+*Here are some standard FAQ to get you started. Feel free to add more!*
+
+* **What are the morals of the story?**
+    * There are very few ways to get insights into running systems. Tracing is state of the art. `console.log` #ftw
+    * Tracing is a static activity and there's no way to dynamically gain insights.
+    * While it's possible to find solutions to these issues, often you don't have insight into if those solutions bring new problems.
+    * Debugging process for non-trivial issues is almost guaranteed to be painful and expensive.
+* **What are the sources for this story?**
+    * [Issue 75](https://github.com/rust-lang/wg-async-foundations/issues/75)
+* **Why did you choose [Barbara] to tell this story?**
+    * Barbara knows what she's doing, but still there is little way to get insights, such as:
```

Comment:
I'd prefer to see this list pulled into its own FAQ, I think

Reply:
Will do that.